### ISSUE TITLE
[Bug]: 1.19 fabric Lags when teleporting between worlds and to distant distances

### ISSUE THREAD

**Issue by MrJorman** [LordDeatHunter/FabricWaystones#135]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://spark.lucko.me/ifdKjkRR9R

When players teleport by Waystones, a lag is obtained.

Players play and when they teleport, such a thing happens, for example, from dimension to dimension. world -> world_the_end or just for a long distance.
shared by all players

![image](https://user-images.githubusercontent.com/55187962/179238034-d3580271-2de0-448b-be71-32c86aa7b798.png)
mods: [message.txt](https://github.com/LordDeatHunter/FabricWaystones/files/9121409/message.txt)

I test one mod (Waystones) in multiplayer and the problem remained

### To Reproduce

1. Teleport between worlds or just over long distances by Waystones

### Expected behavior

So as not to lag and load the server.

### Mod Version

v2.5.0

### Minecraft Version

1.19

### Java Version

Java 17

### Fabric Version

- fabric-loader-0.14.8
- fabric-api-0.57.0+1.19

**Comment by seanxiecn** [comment]
I met this problem, not only the single player but also multi ones. However some times the lags didn't appear, it's weird, and the waystone(fabric) has this problem too.

**Comment by LordDeatHunter** [comment]
I'm not convinced this is an issue with Fabric Waystones. I tested the performance of teleporting via waystones and teleporting using the vanilla Minecraft commands, and the results were pretty much the same. The distance I used for the same-dimension waystones was 10k for both X and Z. With the vanilla commands, I only teleported to the same positions as the waystones, since going any further would cause worldgen lag, which shouldn't happen with normal waystone teleportation.
